src/services/schemas/SchemasService.ts

```diff
@@ -39,12 +39,15 @@ export class SchemaService {
   getProtoSchemaNames(container: ICacheContainer): ng.IPromise<string[]> {
     let deferred: ng.IDeferred<string[]> = this.$q.defer<string[]>();
     this.findTarget(container.serverGroup).then(coordinator => {
-        return this.dmrService.executePost({
+        this.dmrService.executePost({
           address: this.getContainerAddress(container.name, coordinator),
           operation: "get-proto-schema-names"
-        }).then(names => deferred.resolve(names));
+        }).then(
+          (names) => deferred.resolve(names),
+          (error) => deferred.reject(error));
       },
-      error => deferred.reject(error));
+      (error) => deferred.reject(error)
+    );
     return deferred.promise;
```

Comment:
The original code is definitely wrong, but I believe that the second `.then` should be chained after the first, see below:

```typescript
registerProtoSchema(container: ICacheContainer, schema: ISchemaDefinition): ng.IPromise<void> {
    let deferred: ng.IDeferred<void> = this.$q.defer<void>();
    this.findTarget(container.serverGroup)
      .then(coordinator => {
          return this.dmrService.executePost({
            address: this.getContainerAddress(container.name, coordinator),
            operation: "register-proto-schemas",
            "file-names": [schema.fileName],
            "file-contents": [schema.fileContents]
          });
        },
        error => deferred.reject(error))
      .then(() => deferred.resolve(), error => deferred.reject(error));
    return deferred.promise;
  }
```